tests/test_apply.py

```diff
@@ -1262,6 +1262,33 @@ def circuit():
         assert np.allclose(res_sv, expected_sv, atol=tol, rtol=0)
         assert np.allclose(res_probs, expected_prob, atol=tol, rtol=0)
 
+    @pytest.mark.parametrize(
+        "op",
+        [
+            qml.BlockEncode,
+            qml.ctrl(qml.BlockEncode, control=(1)),
+        ],
+    )
+    def test_apply_BlockEncode(self, op, qubit_device, tol):
+        """Test apply BlockEncode and C(BlockEncode)"""
+
+        dev = qubit_device(wires=3)
+        dev_default = qml.device("default.qubit", wires=3)
+
+        def circuit1(A):
+            qml.Hadamard(0)
+            qml.Hadamard(1)
+            qml.BlockEncode(A, wires=[0, 2])
+            op(A, wires=[0, 2])
+            return qml.state()
+
+        A = np.array([[1, 1], [1, -1]]) / np.sqrt(2)
```

Comment:
Could you add notes for `A` here? At first glance, `A` is a matrix of 4 elements but with `2` target wires.

Reply:
I'll add a link to the doc of `qml.BlockEncode` here. Would it be enough? https://docs.pennylane.ai/en/stable/code/api/pennylane.BlockEncode.html